src/app/components/Disclaimer/index.test.tsx

```diff
@@ -43,60 +39,51 @@ const renderComponent = (
         },
       }}
     >
+      {/* @ts-expect-error - to diagnose */}
       <ServiceContext.Provider value={{ disclaimer, externalLinkText }}>
         <DisclaimerComponent />
       </ServiceContext.Provider>
     </ToggleContextProvider>,
   );
-
 describe('Disclaimer Component', () => {
   it('should render a section with role region', () => {
-    const { getByRole } = renderComponent();
-    expect(getByRole('region', { container: 'section' })).toBeInTheDocument();
+    const { container } = renderComponent();
+    const region = screen.getByRole('region');
+    expect(region).toBeInTheDocument();
+    const section = container.querySelector('section');
+    expect(section).toBeInTheDocument();
   });
-
   it('should render disclaimer text correctly', () => {
     const { getByText } = renderComponent();
     expect(
       getByText(
         'Приложение Русской службы BBC News доступно для и . Вы можете также подписаться на наш канал в .',
       ),
     ).toBeInTheDocument();
-
     expect(getByText('IOS')).toBeInTheDocument();
-
     expect(getByText('Android')).toBeInTheDocument();
-
     expect(getByText('Telegram')).toBeInTheDocument();
   });
-
   it('should render links correctly', () => {
     const { getAllByRole } = renderComponent();
     expect(getAllByRole('link').length).toBe(3);
   });
-
   it('should not render the disclaimer when the disclaimer toggle is not enabled', () => {
     const { container } = renderComponent({ enabled: false });
     expect(container).toBeEmptyDOMElement();
   });
-
   it('should not render when disclaimer is null', () => {
+    // @ts-expect-error - not expecting null
```

Comment:
Used comments in this file to stop errors. I think this could be avoided with a better use of types.

Reply:
I think this is OK. If you really wanted to get rid of expected errors, you could remove the test, since typescript is technically preventing you from passing a null value to your component.